File: 240307/키를 기준으로 정렬/sort-by-height.py
class Body:
  def __init__(self,name,height,weight) :
    self.name = name
    self.height = height
    self.weight = weight

n = int(input())

information = []
for i in range(n):
  name, height, weight = tuple(input().split())
  information.append(Body(name, int(height), int(weight)))

information.sort(key= lambda x: x.height)
for j in information:
  print(j.name, j.height, j.weight)

File: 240501/키를 기준으로 정렬/sort-by-height.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split(`\n`)

const n = Number(input[0])

let people = []

class Person {
    constructor(name, height, weight) {
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
}

for (let i = 1; i < n+1; i++) {
    const [name, height, weight] = input[i].split(' ')
    people.push(new Person(name, Number(height), Number(weight)))
}

people.sort((cur, prev) => cur.height - prev.height)

for (let j = 0; j < n; j++) {
    console.log(people[j].name + ' ' + people[j].height + ' ' + people[j].weight)
}